switcher.js

```diff
@@ -201,33 +228,141 @@ class Switcher {
     .then(data => this._handleMissingDefaultProfile(data));
   }
 
+  /**
+   * Parses the config file into an Array of profile objects,
+   * appends the config settings to the existing profile object
+   *
+   * @param {array} Profile objects - Existing array or profile objects
+   * @returns {Promise.<Array>} - Array of profile objects
+   * @reject {Error} - fs.readFile error
+   */
+  _getConfig(profileData) {
```

Comment:
This function is nearly identical to `_getCredentials`, which is fine. It's just that a lot of the logic in here should probably be extracted so that it can be shared between the two functions and not duplicated.